### ISSUE TITLE
PHP runtime: Support for Cloud Logging daemon

### ISSUE THREAD

**Issue by sonnysasaka** [GoogleCloudPlatform/buildpacks#396]
In the [Cloud Logging documentation](https://cloud.google.com/logging/docs/setup/php#gae-config), it's mentioned that app engine flex environment supports the `enable_stackdriver_integration: true` option that enables PHP apps to not be blocked by network due to sending batch logs to Cloud Logging, but instead sends the batch logs locally to another running daemon. However, testing reveals that this is not supported in the flex runtime ever since using this buildpacks (I can only find evidence that this is supported in the previous docker-based runtime https://github.com/GoogleCloudPlatform/php-docker).

Some questions, requests, and recommendations:

1. If this is true, please update the documentation to reflect the case that this is not supported in the buildpacks-based flex environment. Otherwise customers may be disappointed by misleading information.
2. Supporting this flag out of the box may be difficult and cannot be delivered soon, but I think there is a quick workaround that can unblock customers to use this feature very quickly:
* Document how to customize the runtime, e.g. supervisord config to run the background logger, something like:
     ```
     [program:stackdriver-batch-daemon]
     command=php vendor/bin/google-cloud-batch daemon
     autostart=true
     ...
     ```
     And set the environment variable `IS_BATCH_DAEMON_RUNNING=1`.
     I got this from hacking the https://github.com/GoogleCloudPlatform/php-docker and tested that this works. If this is the right way, Google could document this to be a "supported" customization to enable this feature.
* In order for that configuration to work, the missing thing from the PHP runtime is enabling the extensions `sysvmsg`, `sysvsem`, and `sysvshm` (https://github.com/googleapis/google-cloud-php-core/blob/686ffd0bb1328cd2ee963c4444e2ab4b111ba3ba/src/SysvTrait.php#L59). I tested from `php -i` that the current runtime only supports `sysvshm`.

As a customer needing this feature, I can just handle the supervisord config manually for now, but I need support to enable the `sysv*` extensions from Google, as this can only be delivered via the downloadable runtime at `https://dl.google.com/runtimes/*/php/*` (https://github.com/GoogleCloudPlatform/buildpacks/blob/e4b37050cfb2cb0cf4a2e2d9a5356ffd5b8d9f29/pkg/runtime/install.go#L201).

*TL;DR: Would Google be able to satisfy the small request to build PHP with `--enable-sysv*`? If this PHP runtime is also open source, please let me know if I can help contribute to it.* (cc: @jama22 @kennethye1)

**Comment by AidasK** [comment]
Thanks for all the steps! It was pretty easy to get it running, but Daemon logger seems to be slower than the regular batcher. 

What test shows for my local page by sending x500 simple logs:
 - File logger, page loads in 0,552s. Execution of log block itself 0.083s.
 - Batch Logger without daemon 0,627s. . Execution of log block itself 0.133s.
 - Batch Logger with daemon 0,695s. Execution of log block itself 0.148s (for some reason it sometimes spiked 1-3x)

x5000:
 - File logger, page loads in 1,540s. Execution of log block itself 0.877s.
 - Batch Logger without daemon 3.527s. . Execution of log block itself 3.093s
 - Batch Logger with daemon 2,507. Execution of log block itself 2.266s

```php
        [,$time] = Benchmark::value(function () {
            for ($i = 0; $i < 500; $i++) {
                \Log::info('TEST PAGE benchmark' . $i);
            }
        });
        dump(sprintf('%.3fs', $time/1000));
```

Page load time tested with `time curl http://localhost/`, wanted to check if log sending on shutdown method is implemented correctly and as I understand, logs in batch logger are sent after the page closes. 

These tests there not running in total isolation on a separate project, but I can't see the reason to use the daemon approach. Maybe only if you need to see near real time logs with a lower memory footprint, then yes.
